app/models/appointment.rb

```diff
@@ -35,10 +35,14 @@ def self.overdue
   end
 
   def call_result=(new_call_result)
-    if new_call_result == "agreed_to_visit"
+    if new_call_result == 'agreed_to_visit'
```

Comment:
Let's pull this out into proper methods. Using attribute setters way is a code smell.

Reply:
Extracted logic for each path into separate private methods.